requestmanager/executor/executor.go

```diff
@@ -36,28 +37,28 @@ type BlockHooks interface {
 	ProcessBlockHooks(p peer.ID, response graphsync.ResponseData, block graphsync.BlockData) hooks.UpdateResult
 }
 
-// AsyncLoadFn is a function which given a request id and an ipld.Link, returns
-// a channel which will eventually return data for the link or an err
-type AsyncLoadFn func(peer.ID, graphsync.RequestID, ipld.Link, ipld.LinkContext) <-chan types.AsyncLoadResult
+// ReconciledLoader is an interface that can be used to load blocks from a local store or a remote request
+type ReconciledLoader interface {
+	SetRemoteState(online bool)
+	RetryLastOfflineLoad() types.AsyncLoadResult
```

Comment:
we should document thread safety and semantics here - i assume that the intention is this loader can only be used by one client.

Reply:
I feel like something is off with this interface as well:

should this just be a 'retryLastLoad()`?
why does this reconciled loader need to know that the state is online/offline? when it asks the linksystem to load when offline, the linksystem can immediately return an error - in the same way it would if the loader was online, but there was a network failure?

Reply:
the thread safety is relatively well documented in reconciledloader.go itself. This is joy of go's whole "put the interface next to where its used" -- you expect docs in one palce that are elsewhere -- maybe I should put a pointer here.

Re: RetryLastOfflineLoad -- it's named as such cause I thought this was really the only useful case -- trying a load again when you go online. But I agree, a global retry would be more useful and clear. In my mind the operations are as follows:
- offline load -> retry online -- this time wait for remote data and try again
- online load -> retry online -- just try loading from local store (if you were online and loaded successfully, it got saved locally, the remote isn't sending any more data on this particular load, so the only possibility is more local data saved from another source)
- online load -> retry offline -- just load local (and previous remote data was saved)
- offline load -> retry offline -- just load local

A future version could theoretically make an actual request to the network to tell the remote to try the block again, but that's not something I think we tackle now at all.

Reply:
done!